woocommerce/woocommerce-ios
# [Mobile Payments] Add dismiss to feature announcement card

Closes: #7251
part of: #7252 


⛔ Do not merge until b62353706c0cb0c8f93254e7c402ebbecd983f30 is reverted: This is just to make testing easier. (Reverted in https://github.com/woocommerce/woocommerce-ios/pull/7260/commits/78434b8ddee482d1e9cb6e9ec2d32f65e1cb9018)

### Description

When a user dismisses the feature announcement card, we want to store that choice and prevent future display of the same card.

#7252 will allow the user to ask to be reminded later, i.e. start displaying the card again in two weeks time.

This change adds the dismiss button to the recently-added `upsellCardReaders` banner, and supports storing the user's preferences for this dismissal and reminder. It will not yet let the user choose to be reminded, it's only the storage that's involved in this PR.

### Testing instructions


Using a US or Canada store:

1. Navigate to Order Details for an unpaid order, then tap Collect Payment. Or: make a new Simple Payment, and go to the Payment Method screen
2. Observe that the Upsell Card Reader banner is shown, with a x button to dismiss.
3. Tap the x button and observe that the banner is removed
4. Go to another order, and observe that the banner doesn't show there either.
5. Close and restart the app (don't delete it) and return to the order, observe that the banner still doesn't show.

To reset and re-test, either delete and reinstall the app, or use the testing workaround from b62353706c0cb0c8f93254e7c402ebbecd983f30 to clear the stored dismissal. This is not part of the feature, and will be removed before it is merged.

1. Tap the `Card` option on the Payment Method screen to reset the stored preference.
2. Close the order, and open it or any other order again
3. Observe that the banner is shown

### Screenshots


https://user-images.githubusercontent.com/2472348/178263842-5523342f-7c87-4380-830a-5b26b658b784.mp4



---
